app/src/sharedTest/java/org/oppia/android/app/testing/FractionInputInteractionViewTestActivityTest.kt

```diff
@@ -0,0 +1,608 @@
+  // will not be used by user
+  fun testFractionInput_withDivideByZero_errorIsNotDisplayed() {
+    ActivityScenario.launch(FractionInputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_fraction_input_interaction_view))
+      .perform(
+        editTextInputAction.appendText(
+          "1/0"
+        )
+      )
+    onView(withId(R.id.fraction_input_error)).check(matches(withText("")))
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as FractionInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testFractionInput_withDivideByZero_submit_divideByZeroErrorIsDisplayed() {
+    ActivityScenario.launch(FractionInputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_fraction_input_interaction_view))
+      .perform(
+        editTextInputAction.appendText(
+          "1/0"
+        )
+      )
+    closeSoftKeyboard()
+    scrollToSubmitButton()
+    onView(withId(R.id.submit_button)).check(matches(isDisplayed())).perform(click())
+    onView(withId(R.id.fraction_input_error))
+      .check(
+        matches(
+          withText(
+            R.string.fraction_error_divide_by_zero
+          )
+        )
+      )
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as FractionInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testFractionInput_withInvalidCharacter_invalidCharacterErrorIsDisplayed() {
+    ActivityScenario.launch(FractionInputInteractionViewTestActivity::class.java).use {
+      onView(withId(R.id.test_fraction_input_interaction_view))
+        .perform(
+          editTextInputAction.appendText(
+            "."
+          )
+        )
+      onView(withId(R.id.fraction_input_error))
+        .check(
+          matches(
+            withText(
+              R.string.fraction_error_invalid_chars
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as FractionInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testFractionInput_withLong_submit_numberTooLongErrorIsDisplayed() {
+    ActivityScenario.launch(FractionInputInteractionViewTestActivity::class.java).use {
+      onView(withId(R.id.test_fraction_input_interaction_view))
+        .perform(
+          editTextInputAction.appendText(
+            "12345678"
+          )
+        )
+      closeSoftKeyboard()
+      scrollToSubmitButton()
+      onView(withId(R.id.submit_button)).check(matches(isDisplayed())).perform(click())
+      onView(withId(R.id.fraction_input_error))
+        .check(
+          matches(
+            withText(
+              R.string.fraction_error_larger_than_seven_digits
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  @DisableAccessibilityChecks // Disabled, as FractionInputInteractionViewTestActivity is a test file and
+  // will not be used by user
+  fun testFractionInput_noInput_submit_errorIsDisplayed() {
```

Comment:
Nit: More descriptive.
```suggestion
  fun testFractionInput_emptyInput_submit_errorIsDisplayed() {
```

Reply:
Done